modules/dists/BlockDist.chpl

```diff
@@ -1538,9 +1538,62 @@ where this.sparseLayoutType == unmanaged DefaultDist &&
   return true;
 }
 
+proc BlockArr.canDoOptimizedSwap(other) {
+  var domsMatch = true;
+  var domShapesMatch = true;
+
+  if this.dom != other.dom { // no need to check if this is true
+    if domsMatch {
+      for param i in 0..this.dom.rank-1 {
```

Comment:
This doesn't need to be a `param`, right?  Is your thought that in the typical case, avoiding the loop will give an execution-time benefit that makes the increase in code size worthwhile?  (or even that the code won't increase much in size?)

Reply:
Yeah you're right. It is my wishful thinking that having it `param` or not doesn't show any actual difference in either compilation or execution time. 

The main reason I have semi-automatically made this loop `param` is that the Block distribution code has several loops over ranks and all of which are `param` loops. (At least the ones I can search and find).

If you feel non-`param` is safer, I can change it for sure.

Reply:
I don't know that I have a preference, I was curious as much as anything.  I think that if it were on a more obviously performance critical path (like local dsiAccess) I'd definitely use a param loop.  Whereas for a distributed memory swap, it's probably not a big win (but also may not be a big generated code size hit.  This one isn't obvious to me which it should be.  If we hadn't been talking about premature optimization of generated code size recently, I might not have even noticed it.

Reply:
I see. Consistency in coding style outweighs my fear of growing the AST size because of the unrolling that happens here. As you don't have strong opinions, I am leaning towards keeping this loop a `param` for now.